### PR TITLE
SIR-2361 Polkadot transfer validation

### PR DESCRIPTION
**Pull request by avolkov-dev**

### CODE REVIEW

**Review by aushakov — APPROVED** [review]

File: src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                            "Invalid networkType: %s, expected %s",
+                            transfer.getBlockchain().getNetworkType(), networkType));
+        }
+
+        BuiltTx builtTx;
+
+        try{
+            builtTx = jsonSerializer.deserialize(new String(unsignedTransaction, StandardCharsets.UTF_8), BuiltTx.class);
+        }catch (Exception e){
+            return TransactionValidationResult.CreateInvalid("Unexpected format " + e.getMessage());
```
Comment (KonstantinRyazantsev) [inline src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java:54]: let's log the exception as an error

File: src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        final var reader = new ScaleCodecReader(Utils.hexToU8a(builtTx.PreparedSigningPayload));
+        final var balanceTransferReader = new BalanceTransferReader(SS58Type.Network.from(PolkadotNetworkTypeHelper.GetPrefix(NetworkType.Private)));
+        final var decoded = balanceTransferReader.read(reader);
+
+        final var amount = PolkadotNetworkTypeHelper.ConvertFromPlank(decoded.getBalance().getValue(), networkType);
+        if (transfer.getValue().getAmount().compareTo(amount) < 0) {
```
Comment (KonstantinRyazantsev) [inline src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java:62]: why not `!= 0` ?

File: src/main/java/io/swisschain/crypto/utils/polkadot/PolkadotNetworkTypeHelper.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    public static BigDecimal ConvertFromPlank(BigInteger source, NetworkType networkType){
+      int decimalPlaces;
+        switch (networkType){
+            case Private:
+            case Test:
+                decimalPlaces= PolkadotNetworkTypeHelper.westendDecimalPlaces;
+                break;
+            default:
+                decimalPlaces= PolkadotNetworkTypeHelper.polkadotPrefix;
```
Comment (KonstantinRyazantsev) [inline src/main/java/io/swisschain/crypto/utils/polkadot/PolkadotNetworkTypeHelper.java:41]: should be `polkadotDecimalPlaces` I guess

File: src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java (hunk 1 above)
Comment (avolkov-dev) [reply src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java:54]: It should be logger on higher level, i assume

File: src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java (hunk 2 above)
Comment (avolkov-dev) [reply src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java:62]: TBH i don't know. I've just done it same way as in other blockchains
See:
[BTC](https://github.com/swisschain/Sirius.Vault.HSM/blob/master/src/main/java/io/swisschain/crypto/transactions/validators/bitcoin/BitcoinTransferTransactionValidator.java#L79)
[ETH](https://github.com/swisschain/Sirius.Vault.HSM/blob/master/src/main/java/io/swisschain/crypto/transactions/validators/ethereum/BaseEthereumTransferTransactionValidator.java#L53)

File: src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java (hunk 1 above)
Comment (KonstantinRyazantsev) [reply src/main/java/io/swisschain/crypto/transactions/validators/polkadot/PolkadotTransferTransactionValidator.java:54]: but the exception is not rethrown here

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**